homeassistant/components/device_automation/toggle_entity.py

```diff
@@ -151,15 +151,12 @@ async def async_attach_trigger(
     """Listen for state changes based on configuration."""
     trigger_type = config[CONF_TYPE]
     if trigger_type == CONF_TURNED_ON:
-        from_state = "off"
         to_state = "on"
     else:
-        from_state = "on"
         to_state = "off"
     state_config = {
         CONF_PLATFORM: "state",
         state_trigger.CONF_ENTITY_ID: config[CONF_ENTITY_ID],
-        state_trigger.CONF_FROM: from_state,
```

Comment:
Why do we remove this?

Reply:
The trigger should fire when the entity changes states TO a certain state, we don't care what state it was before.
Background here: https://github.com/home-assistant/core/pull/45127